projects/js-packages/charts/src/providers/theme/theme-provider.tsx

```diff
@@ -46,8 +29,7 @@ type ThemeProviderProps = {
 // Provider component for chart theming
 // Allows theme customization through props while maintaining default values
 const ThemeProvider: FC< ThemeProviderProps > = ( { theme = {}, children } ) => {
-	const mergedTheme = { ...defaultTheme, ...theme };
-	return <ThemeContext.Provider value={ mergedTheme }>{ children }</ThemeContext.Provider>;
+	return <ThemeContext.Provider value={ theme }>{ children }</ThemeContext.Provider>;
```

Comment:
Change this to a partial theme and stop merging with default, so that if defined it can override the global theme, which is already based on default.